### PR TITLE
Update FindACE.cmake

### LINKED ISSUE #340: FindACE.cmake is broken if invoked multiple times
This snippet of code:

```cmake
find_package(ACE)
find_package(ACE)

```

generates this error:

```bash
CMake Error at /home/ngenesio/robotology/robotology-superbuild/build/install/share/YCM/find-modules/FindACE.cmake:170 (add_library):
  add_library cannot create imported target "ACE::ACE" because another target
  with the same name already exists.
Call Stack (most recent call first):
  CMakeLists.txt:28 (find_package)


CMake Error at /home/ngenesio/robotology/robotology-superbuild/build/install/share/YCM/find-modules/FindACE.cmake:192 (add_library):
  add_library cannot create imported target "ACE::ACE_INLINE" because another
  target with the same name already exists.
Call Stack (most recent call first):
  CMakeLists.txt:28 (find_package)
```

### PR DESCRIPTION
**Pull request by Nicogene**
It fixes #340

### CODE REVIEW

**Comment by traversaro** [comment]
For consistency with CMake's provided FindModules (see https://gitlab.kitware.com/cmake/cmake/-/blob/v3.16.8/Modules/FindJPEG.cmake#L108 or https://gitlab.kitware.com/cmake/cmake/-/blob/v3.16.8/Modules/FindPNG.cmake#L107), I would not modify the already existing imported target if it is already defined.

**Comment by drdanz** [comment]
Also please rebase over the ycm-0.11 branch and add a line to the release notes

File: help/release/0.11.2.rst [hunk 1]
```diff
@@ -14,3 +14,4 @@ Modules
 ---------
 
 * Fixed :module:`FindOctave` for Octave 5
+* Fixed :module:`FindACE`(see #340)
```
Comment (drdanz) [inline help/release/0.11.2.rst:17]: Sorry for being pedantic, but `Fixed FindACE` does not mean anything for a person reading the changelog. Please write something useful also here (feel free to write the same you wrote in the commit log)

**Comment by drdanz** [comment]
Merged, thanks.

**Pull request merged**